exercises/reforestation/canonical-data.json

```diff
@@ -0,0 +1,93 @@
+{
+    "exercise": "reforestation",
+    "version": "1.0.0",
+    "cases": [
+      {
+        "description": "Base case: empty tree",
+        "property": "treeTraversals",
+        "input": {
+          "preorder": [],
+          "inorder": []
+        },
+        "expected": {}
+      },
+      {
+        "description": "1 more step than base case",
+        "property": "treeTraversals",
+        "input": {
+            "preorder": ["a"],
+            "inorder": ["a"]
+        },
+        "expected": {"v":"a", "l": {}, "r": {}}
+      },
+      {
+        "description": "Core functionality with realistic size traversal",
+        "property": "treeTraversals",
+        "input": {
+            "preorder": ["a","i","x","f","r"],
+            "inorder": ["i","a","f","x","r"]
+        },
+        "expected": {"v": "a",
+            "l": {"v": "i", "l": {}, "r": {}},
+            "r": {"v": "x",
+                "l": {"v": "f", "l": {}, "r": {}},
+                "r": {"v": "r", "l": {}, "r": {}}
+            }
+        }
+      },
+      {
+        "description": "Traversals must be fully consumed. No extra values.",
+        "property": "treeTraversals",
+        "input": {
+          "preorder": ["a", "b"],
+          "inorder": ["b", "a", "r"]
+        },
+        "expected": {"error": "invalid traversal"}
+      },
+      {
+          "description": "Reject inconsistent traversals of same length",
+          "property": "treeTraversals",
+          "input": {
+              "preorder": ["x","y","z"],
+              "inorder": ["a","b","c"]
+          },
+          "expected": {"error": "traversals are not consistent"}
+      },
+      {
+        "description": "BonusPoints: generate multiple trees",
```

Comment:
I don't really like the "BonusPoints: " prefix. Also, it is hard to create a test generator for the current test data, as this test case expects an array to be returned for the `treeTraversals` property, whereas the other test cases return an object and not an array. I think there are a couple of options:

- Have all test cases return an array (which would have length equal to one for the previous test cases)
- Name the property differently for this test case
- Remove this additional test case

Reply:
This is the second suggestion to remove this test case. I have been convinced to remove it.

Reply:
I think you can add it as a Prolog-specific extension to the exercise, since I believe multiple results are generated somewhat automatically in case of duplicates if you choose the right approach?

I'd add this as a hint; you can always add a track-specific test, too.

(I assume that Prolog doesn't have test-generating scripts, so the overhead of adding custom tests in combination with deriving unit tests from canonical data need not be addressed in your track's case.)